platform/android/os_android.cpp

```diff
@@ -222,6 +222,27 @@ String OS_Android::get_user_data_dir() const {
 	return ".";
 }
 
+String OS_Android::get_external_data_dir() const {
+	String data_dir = godot_io_java->get_external_data_dir();
+	if (data_dir != "") {
+		// Workaround for Android 6.0+ using a symlink.
+		// Save the current directory.
+		char current_dir_name[2048];
+		getcwd(current_dir_name, 2048);
+		// Change directory to the external data directory.
+		chdir(data_dir.utf8().get_data());
+		// Get the actual directory without the potential symlink.
+		char data_current_dir_name[2048];
+		getcwd(data_current_dir_name, 2048);
+		// Convert back to a String.
+		data_dir.parse_utf8(data_current_dir_name);
+		// Restore original current directory.
+		chdir(current_dir_name);
+		return data_dir;
+	}
```

Comment:
This is the same logic as in `get_user_data_dir()`. Can you wrap the logic into a util function and have both functions call it instead.

Reply:
Done.